test/checking.plt

```diff
@@ -11,25 +11,39 @@
-                           source: "prolog_xref"},
-                         _{message: "Singleton variable Gae",
-                           range: _{start: _{line: 1, character: 4}, end: _{line: 1, character: 7}},
-                           severity: 2,
-                           source: "prolog_xref"},
-                         _{message: "Singleton variable B",
-                           range: _{start: _{line: 3, character: 9}, end: _{line: 3, character: 10}},
-                           severity: 2,
-                           source: "prolog_xref"},
-                         _{message: "Singleton variable Z",
-                           range: _{start: _{line: 4, character: 11}, end: _{line: 4, character: 12}},
-                           severity: 2,
-                           source: "prolog_xref"}] ) ]) :-
+     [ true( Errors =@=  [_{ message: "Operator expected",
+                              range: _{ end: _{ character: 0, line: 10 },
+                                      start: _{ character: 11, line: 9 } },
+                              severity: 1,
+                              source: "prolog_xref" },
+                          _{ message: "Singleton variable Gae",
+                              range: _{ end: _{ character: 7, line: 4 },
+                                      start: _{ character: 4, line: 4 } },
+                              severity: 2,
+                              source: "prolog_xref" },
+                          _{ message: "Singleton variable B",
+                              range: _{ end: _{ character: 10, line: 6 },
+                                      start: _{ character: 9, line: 6 } },
+                              severity: 2,
+                              source: "prolog_xref" },
+                          _{ message: "Singleton variable Z",
+                              range: _{ end: _{ character: 12, line: 7 },
+                                      start: _{ character: 11, line: 7 } },
+                              severity: 2,
+                              source: "prolog_xref" },
+                          _{ message: "Module `nexiste_pas` not found",
+                              range: _{ end: _{ character: 25, line: 1 },
+                                      start: _{ character: 14, line: 1 } },
+                              severity: 2,
+                              source: "prolog_xref" },
+                          _{ message: "Module `library(does_not_exist)` not found",
+                              range: _{ end: _{ character: 37, line: 0 },
+                                      start: _{ character: 14, line: 0 } },
+                              severity: 2,
+                              source: "prolog_xref" } ]) ]) :-
     context_module(ThisModule),
     module_property(ThisModule, file(ThisFile)),
     relative_file_name(InputFile, ThisFile, './checking_input1.pl'),
-    check_errors(InputFile, Errors).
+    once(check_errors(InputFile, Errors)).
```

Comment:
Rather than needing to add `once/1` here, can we make `check_errors/2` deterministic? If you run the query in the top-level and get a choice-point (where you'd normally press `.` to end or `;` to see the next), you can press `*` to see where the choice points are. From a quick glance, I suspect the second call to `member/2` in `usemod_filespec_position/4` - could that be `memberchk/2`?

Reply:
I'm honestly not sure why I added `once` here 🤦🏻‍♂️  I've poked around and I can't find a case where `check_errors` or `usemod_filespec_position` leaves a choice-point. I must have added `once` while I was testing stuff out and forgot to remove it.